docs/api.md

```diff
@@ -589,3 +589,241 @@ state.
 
 Please note only the existence of the annotation is important to treat the BMH
 as detached and the value of the annotation is always ignored.
+
+## HostFirmwareSettings
+
+A **HostFirmwareSettings** resource is used to manage BIOS settings for a host,
+there is a one-to-one mapping with **BareMetalHosts**.  A
+**HostFirmwareSettings** resource is created when BIOS settings are read from
+Ironic as the host moves to the Ready state.  These settings are the complete
+actual BIOS configuration names returned from the BMC, typically 100-200
+settings per host, as compared to the three vendor-independent fields stored in
+the **BareMetalHosts** `firmware` field.
+
+### HostFirmwareSettings spec
+
+The *HostFirmwareSettings's* *spec* defines the desired BIOS settings. These
+settings will be sent to Ironic as part of clean-steps for update to the BMC
+when the host goes through cleaning.
```

Comment:
Probably useful to refer the BMO states also, just for the sake of clarity

Reply:
Good point, fixed.